services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,43 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (event != NULL) {
+    if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == pdPASS)
+      return ERR_CODE_SUCCESS;
+    else
+      return ERR_CODE_QUEUE_FULL;
+  }
+  else
+    return ERR_CODE_INVALID_ARG;
 
-  return ERR_CODE_SUCCESS;
+  return ERR_CODE_INVALID_QUEUE_MSG;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT_CMD;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t eventBuf;
+
   while (1) {
-    
+
+    float temp;
+    error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
receive and check the event first before reading temp, since right now even if you don't receive a measure temp command you'll be reading the temperature